sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -1424,11 +1459,14 @@ private void SetNameFieldsIfNull()
                 {
                     scope.Start();
 
-                    if (UsingClientSideEncryption)
+                    if (UsingClientSideEncryption && !ignoreClientSideEncryption)
                     {
                         // TODO #27253
                         //options = BlobDownloadOptions.CloneOrDefault(options) ?? new BlobDownloadOptions();
-                        range = BlobClientSideDecryptor.GetEncryptedBlobRange(range);
+                        if ((await GetPropertiesInternal(conditions, async, cancellationToken).ConfigureAwait(false)).Value.Metadata.TryGetValue(Constants.ClientSideEncryption.EncryptionDataKey, out string rawEncryptiondata))
+                        {
+                            range = BlobClientSideDecryptor.GetEncryptedBlobRange(range, rawEncryptiondata);
+                        }
                     }
```

Comment:
is there other reason than get properties call for `ignoreClientSideEncryption` ? 
It seems that we're pushing up implementation to callers where properties are already available.

Would passing optional properties/metadata to this method eliminate need for  `ignoreClientSideEncryption` flag? I.e. call getProperties if needed?

Reply:
It ignores overlapping ranges for something like a partitioned download, and also just skips decryption as a whole. This was trivial in v1 when ranges were extended a handful of bytes, but now we are extending to ~4MB bounds, which may not play nice with transfer options. While we could pass in optional prefetched properties, we also ignore potential boundary extension waste by just shifting decryption back to the partitioned downloader.

Reply:
I do see how just adding bool flags is unfun, though. Perhaps it's better for the "real" DownloadStreamingInternal method as it exists today to not have a concept of client-side encryption. Instead, the public ranged download APIs go through a "DirectCalledDownloadStreamingInternal" which applies the appropriate transformations. All customer-called ranged downloads go through there, and our partitioned downloader and open-read don't. This way every public API goes through some function that applies client-side encryption, but we don't have flagged branching logic in our low-level REST download. 

Thoughts?

Reply:
yeah, that'd be my next suggestion. `UsingClientSideEncryption` and `ignoreClientSideEncryption` in same condition looks odd and smells like spaghetti.